### PR TITLE
[WIP]Access http response

### PR DESCRIPTION
**Pull request by lightsprint09**
Fixes issues:  #49 , #39

New feature:
When doing a network request, you may want to access the raw HTTPResponse. In our current implementation is this not possible. This opens up the API to access the HTTPResponse when needed.

#### Make sure to check all boxes before merging

- [x] Method/Class documentation
- [x] README.md documentation
- [x] Unit tests for new features/regressions

### CODE REVIEW

**Comment by dennispost** [comment]
@lightsprint09 I'm missing a description of the PR goals

File: Source/NetworkServiceMock.swift [hunk 1]
```diff
@@ -84,9 +84,9 @@ public class NetworkServiceMock: NetworkServiceProviding {
     /// - Parameters:
     ///   - data: the mock response from the server. `Data()` by default
     ///   - count: the count how often the response gets triggerd. 1 by default
```
Comment (dennispost) [inline Source/NetworkServiceMock.swift]: Missing httpResponse parameter doc

File: Source/NetworkServiceMock.swift [hunk 2]
```diff
@@ -98,9 +98,9 @@ public class NetworkServiceMock: NetworkServiceProviding {
...
```
Comment (dennispost) [inline Source/NetworkServiceMock.swift]: Missing httpResponse parameter doc

File: Source/RetryNetworkService.swift [hunk 3]
```diff
@@ -61,15 +61,15 @@ public final class RetryNetworkService: NetworkServiceProviding {
     }
     
     @discardableResult
```
Comment (dennispost) [inline Source/RetryNetworkService.swift]: Missing function documentation

Comment (lightsprint09) [inline Source/RetryNetworkService.swift]: This is a protocol implementation. So the method is documented by the protocol

File: Source/NetworkResponseProcessor.swift [hunk 4]
```diff
@@ -94,15 +80,16 @@ extension NetworkResponseProcessing {
...
+                    onError(DBNetworkStackError.unknownError)
+                }
             }
-        } catch let parsingError as DBNetworkStackError {
-            queue.async {
-                return onError(parsingError)
-            }
-        } catch {
+        } catch let parsingError {
+            let dbBetworkError: DBNetworkStackError! = parsingError as? DBNetworkStackError
```
Comment (ch-one) [inline Source/NetworkResponseProcessor.swift]: `dbBetworkError` => `dbNetworkError`

Comment (lightsprint09) [inline Source/NetworkResponseProcessor.swift]: 🦇 -Work??

File: Source/NetworkServiceProviding.swift [hunk 5]
```diff
@@ -33,11 +33,11 @@ import Dispatch
  */
 public protocol NetworkServiceProviding {
     /**
-     Fetches a resource asynchrony from remote location
+     Fetches a resource asynchrony from remote location.
```
Comment (dennispost) [inline Source/NetworkServiceProviding.swift]: Do you mean 'asynchronously'?

File: Source/NetworkServiceProviding.swift [hunk 6]
```diff
@@ -49,9 +49,20 @@ public protocol NetworkServiceProviding {
 
 public extension NetworkServiceProviding {
     /**
-     Fetches a resource asynchrony from remote location
+     Fetches a resource asynchrony from remote location. Completion and Error block will be called on the main thread.
```
Comment (dennispost) [inline Source/NetworkServiceProviding.swift]: See above

File: Source/NetworkServiceProviding.swift [hunk 7]
```diff
@@ -49,9 +49,20 @@ public protocol NetworkServiceProviding {
...
+
```
Comment (dennispost) [inline Source/NetworkServiceProviding.swift]: Demo code should be part of Playground

**Pull request merged**